components/dashboard/src/settings/SelectIDE.tsx

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { IDEOption, IDEOptions } from "@gitpod/gitpod-protocol/lib/ide-protocol";
+import { useContext, useEffect, useState } from "react";
+import InfoBox from "../components/InfoBox";
+import SelectableCardSolid from "../components/SelectableCardSolid";
+import Tooltip from "../components/Tooltip";
+import { getGitpodService } from "../service/service";
+import { UserContext } from "../user-context";
+import CheckBox from "../components/CheckBox";
+import { User } from "@gitpod/gitpod-protocol";
+
+interface SelectIDEProps {
+    updateUserContext?: boolean;
+}
+
+export const updateUserIDEInfo = async (user: User, selectedIde: string, useLatestVersion: boolean) => {
+    const additionalData = user?.additionalData ?? {};
+    const settings = additionalData.ideSettings ?? {};
+    settings.settingVersion = "2.0";
+    settings.defaultIde = selectedIde;
+    settings.useLatestVersion = useLatestVersion;
+    additionalData.ideSettings = settings;
+    getGitpodService()
+        .server.trackEvent({
+            event: "ide_configuration_changed",
+            properties: settings,
+        })
+        .then()
+        .catch(console.error);
+    return getGitpodService().server.updateLoggedInUser({ additionalData });
+};
+
+export default function SelectIDE(props: SelectIDEProps) {
+    const { user, setUser } = useContext(UserContext);
+
+    // Only exec once when we access this component
+    useEffect(() => {
+        user && User.migrationIDESettings(user);
+    }, []);
+
+    const actualUpdateUserIDEInfo = async (user: User, selectedIde: string, useLatestVersion: boolean) => {
+        const newUserData = await updateUserIDEInfo(user, selectedIde, useLatestVersion);
+        props.updateUserContext && setUser({ ...newUserData });
```

Comment:
@andreafalzetti and @mustard-mh, please explain this, it's not clear what `updateUserContext` is for here. it looks like a switch to make changes on backend but not to reflect on frontend, it's hard to see why this is wanted w.r.t to UX.

Reply:
🙏  Will do this in following up PR that works for issue https://github.com/gitpod-io/gitpod/issues/6707